content/en/docs/concepts/storage/volumes.md

```diff
@@ -11,228 +11,193 @@ weight: 10
+Volumes can not mount into other volumes, or have hard links to
+other volumes. Each container in the pod's configuration must independently specify where to mount each volume. 
 
-### azureDisk {#azuredisk}
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: sample-cache-pod
+spec:
+  volumes:
+    [...]
+  containers:
+    - name: test
+      image: busybox
+      volumeMounts:
+        - name: cache-volume
+          mountPath: /var/cache
 
-The `azureDisk` volume type mounts a Microsoft Azure [Data Disk](https://docs.microsoft.com/en-us/azure/aks/csi-storage-drivers) into a pod.
+```
 
-For more details, see the [`azureDisk` volume plugin](https://github.com/kubernetes/examples/tree/{{< param "githubbranch" >}}/staging/volumes/azure_disk/README.md).
+### Core volume types {#volume-types-core}
 
-#### azureDisk CSI migration
+Kubernetes supports several types of volumes. These core volume types are 
+integrated into Kubernetes, and include network storage volume types and storage provided by the pod itself. 
 
-{{< feature-state for_k8s_version="v1.19" state="beta" >}}
+Core Volume Types:
+   * [cephfs](#cephfs)
+   * [configMap](#configmap)
+   * [downwardAPI](#downwardapi)
+   * [emptyDir](#emptydir)
+   * [fc (fibre channel)](#fc)
+   * [gitRepo (deprecated)](#gitrepo)
+   * [glusterfs](#glusterfs)
+   * [hostPath](#hostpath)
+   * [iscsi](#iscsi)
+   * [local](#local)
+   * [nfs](#nfs)
+   * [projected](#projected)
+   * [rbd](#rbd)
+   * [secret](#secret)
 
-The `CSIMigration` feature for `azureDisk`, when enabled, redirects all plugin operations
-from the existing in-tree plugin to the `disk.csi.azure.com` Container
-Storage Interface (CSI) Driver. In order to use this feature, the [Azure Disk CSI
-Driver](https://github.com/kubernetes-sigs/azuredisk-csi-driver)
-must be installed on the cluster and the `CSIMigration` and `CSIMigrationAzureDisk`
-features must be enabled.
 
-### azureFile {#azurefile}
+### Storage plugins
 
-The `azureFile` volume type mounts a Microsoft Azure File volume (SMB 2.1 and 3.0)
-into a pod.
+A [persistentVolumeClaim](#persistentvolumeclaim) is a declarative way to ask for storage, without preference for the underlying implementation. 
 
-For more details, see the [`azureFile` volume plugin](https://github.com/kubernetes/examples/tree/{{< param "githubbranch" >}}/staging/volumes/azure_file/README.md).
+This is powered by storage plugins. More specifically, Kubernetes implements the {{< glossary_tooltip text="Container Storage Interface" term_id="csi" >}} (CSI),
+which lets developers implement storage plugins without needing to make changes to the code of
+Kubernetes itself. Visit
+[Kubernetes Container Storage Interface Documentation])(https://kubernetes-csi.github.io/docs/)
+to learn more about CSI. 
 
-#### azureFile CSI migration
+[FlexVolume](#flexvolume) is an another mechanism for storage plugins. Unlike CSI, FlexVolume is
```

Comment:
We should say this is deprecated and CSI is the recommended way to develop storage plugins for Kubernetes.